cl/audio/feeds.py

```diff
@@ -124,22 +162,37 @@ def items(self, obj):
         search_form = SearchForm(obj.GET)
         if search_form.is_valid():
             cd = search_form.cleaned_data
-            with Session() as session:
-                solr = ExtraSolrInterface(
-                    settings.SOLR_AUDIO_URL, http_connection=session, mode="r"
-                )
-                main_params = search_utils.build_main_query(
-                    cd, highlight=False, facet=False
-                )
-                main_params.update(
-                    {
-                        "sort": "dateArgued desc",
-                        "rows": "20",
-                        "start": "0",
-                        "caller": "SearchFeed",
-                    }
+            if waffle.flag_is_active(obj, "oa-es-deactivate"):
+                with Session() as session:
+                    solr = ExtraSolrInterface(
+                        settings.SOLR_AUDIO_URL,
+                        http_connection=session,
+                        mode="r",
+                    )
+                    main_params = search_utils.build_main_query(
+                        cd, highlight=False, facet=False
+                    )
+                    main_params.update(
+                        {
+                            "sort": "dateArgued desc",
+                            "rows": "20",
+                            "start": "0",
+                            "caller": "SearchFeed",
+                        }
+                    )
+                    items = solr.query().add_extra(**main_params).execute()
+                    return items
+            else:
+                override_params = {
+                    "order_by": "dateArgued desc",
+                }
+                search_query = AudioDocument.search()
+                items = do_es_podcast_query(
+                    search_query,
+                    cd,
+                    override_params,
+                    rows=20,
```

Comment:
It seems a lot tidier to update `cd` here, rather than in the function via `override_params`. If you do it here, you create a complete package of params to send, then do so. If you do it in the function (as it is now), you create two packages of params here, which you merge elsewhere.

Reply:
Yeah, this is clearer and tidier, so we only pass `cd` as parameter, changed.